### ISSUE TITLE
Bug? NaN loss after training for a while using for BERT Encoded sentences.

### ISSUE THREAD

**Issue by codeninja** [huggingface/transformers#3355]
# ❓ Questions & Help

## Details

I have a model which is taken from the HF Examples and slightly modified.

```
from transformers import TFBertModel, TFBertForSequenceClassification, BertTokenizer
# configuration = BertConfig()
def build_bert(batch_size=1, use_logits=True):
    num_labels = max(max(train_label), max(test_label))
    print(f"Number of labels: {num_labels}")

    optimizer = tf.keras.optimizers.Adam(learning_rate=1e-5, epsilon=1e-06, clipnorm=1.0)

    if num_labels == 1:
        loss = tf.keras.losses.MeanSquaredError()
    else:
        loss = tf.keras.losses.SparseCategoricalCrossentropy(from_logits=use_logits)

    print(f"loss used: {loss}")
    macc = tf.keras.metrics.SparseCategoricalAccuracy('accuracy')
    vacc = tf.keras.metrics.SparseCategoricalAccuracy('val_accuracy')

    config = BertConfig.from_pretrained("bert-base-cased", num_labels=num_labels, batch_size=batch_size)
    bert_model = TFBertForSequenceClassification.from_pretrained("bert-base-cased", config=config)
    bert_model.compile(optimizer=optimizer, loss=loss, metrics=[macc, vacc])
    return bert_model
```

I am training on data which is encoded by BERT into 48 token arrays and encoded with the HF Bert Encoder.

```bert_model = build_bert(1000, False)
bert_model.fit([encodings[30000:40000], train_attn_mask[30000:40000]], classes[30000:40000], 
               epochs=1, validation_split=.1, shuffle=False)
```

![image](https://user-images.githubusercontent.com/14914/77113444-d683f700-69f8-11ea-8f4e-b9991e777b16.png)

My model will train for a while (Please pardon the output... I have no idea why jupyter lab does this.)
![image](https://user-images.githubusercontent.com/14914/77113488-ee5b7b00-69f8-11ea-9029-fdda4ec26a31.png)

Then at some point (different every run) the loss drops to NaN.
![image](https://user-images.githubusercontent.com/14914/77113529-016e4b00-69f9-11ea-94d4-010388ff0f35.png)

**A link to original question on Stack Overflow**: 
The solution for this problem on SO is varied. I tried changing the optimizer learning rates as well as altering the epsilon. I have validated that my data does not contain nan values, negative classifications,  or invalid encodings. I have removed all non unicode characters.

My concern is that this has uncovered a bug with in the framework.

**Comment by LeiYangJustin** [comment]
Do you solve the problem?
